library/src/main/java/org/xmtp/android/library/Conversations.kt

```diff
@@ -148,6 +153,31 @@ data class Conversations(
         } ?: emptyList()
     }
 
+    private fun validateConsentSignature(signature: String, clientAddress: String, peerAddress: String, timestamp: Long): Boolean {
+        val messageData = Signature.newBuilder().build().consentProofText(peerAddress, timestamp).toByteArray()
+        val signatureData = Numeric.hexStringToByteArray(signature)
+        val sig = Signature.parseFrom(signatureData)
+        val recoveredPublicKey = KeyUtil.recoverPublicKeyKeccak256(sig.rawData.toByteString().toByteArray(), Util.keccak256(messageData))
+            ?: return false
+        return clientAddress == KeyUtil.publicKeyToAddress(recoveredPublicKey)
+    }
```

Comment:
I kind of feel like this belongs in the KeyUtil class instead.

Reply:
Updated!